src/core/Global.ttslua

```diff
@@ -283,6 +284,134 @@ function getTokenIndex(token)
+  -- cards that dictate only some tokens are eligible for replacement
+    for _, token in ipairs(tokensInPlay) do
+      if string.find(dataFromLastReplacementEffect.tokenList, token.getName()) then
+        matchingTokensInPlay[#matchingTokensInPlay + 1] = token
+      end
+    end
+  else
+  -- for any cards that can replace any tokens
+    matchingTokensInPlay = tokensInPlay
+  end
+
+  if #matchingTokensInPlay == 0 then
+    broadcastToAll("No eligible token found in play area", "Red")
+    return
+  end
+
+  if #matchingTokensInPlay > 1 then
+    for _, token in ipairs(matchingTokensInPlay) do
+      -- draw XML to return token to bag
+      token.UI.setXmlTable({
+        {
+          tag = "VerticalLayout",
+          attributes = {
+            height = 275,
+            width = 275,
+            spacing = 0,
+            padding = "0 0 20 25",
+            scale = "0.4 0.4 1",
+            rotation = "0 0 180",
+            position = "0 0 -15",
+            color = "rgba(0,0,0,0.7)",
+            onClick = "Global/returnAndRedraw(" .. token.getGUID() .. ")",
+          },
+          children = {
+            {
+              tag = "Text",
+              attributes = {
+                fontSize = "100",
+                font = "font_teutonic-arkham",
+                color = "#ffffff",
+                text = "Redraw"
+              }
+            },
+            {
+              tag = "Text",
+              attributes = {
+                fontSize = "125",
+                font = "font_arkhamicons",
+                color = "#ffffff",
+                text = "u"
+              }
+            }
+          }
+        }
+      })
+    end
+    isTokenXMLActive = true
+  -- no need to make buttons if there is only one eligible token to return and redraw
+  else
+    returnAndRedraw(_, matchingTokensInPlay[1].getGUID())
+  end
+end
+
+function returnAndRedraw(_, tokenGUID)
+  local indexOfReturnedToken
+  local takeParameters = {}
+  local returnedToken = getObjectFromGUID(tokenGUID)
+  local matColor = playmatApi.getMatColorByPosition(returnedToken.getPosition())
+  local mat = guidReferenceApi.getObjectByOwnerAndType(matColor, "Playermat")
+  
+  isTokenXMLActive = false
+  trackChaosToken(returnedToken.getName(), mat.getGUID(), true)
+  indexOfReturnedToken = getTokenIndex(returnedToken)
+  takeParameters.position = returnedToken.getPosition()
+
+  if #chaosTokens > indexOfReturnedToken then
+    takeParameters.rotation = mat.getRotation() + Vector(0, 0, -8)
+  else
+    takeParameters.rotation = returnedToken.getRotation()
+  end
+
+  returnChaosTokenToBag(returnedToken)
+  if dataFromLastReplacementEffect.redrawnTokenType == "random" then
+    token = drawChaosToken({mat = mat, drawAdditional = true, takeParameters = takeParameters})
+  else
+    token = drawChaosToken({mat = mat, drawAdditional = true, takeParameters = takeParameters, tokenType = dataFromLastReplacementEffect.redrawnTokenType})
```

Comment:
If you change `redrawnTokenType` to a boolean field like `drawRandomReplacement = true / false` then you don't need this whole if statement